frontend/packages/console-app/src/components/user-preferences/UserPreferences.tsx

```diff
@@ -0,0 +1,157 @@
+
+  const [userPreferencesItemExtensions, userPreferencesItemResolved] = useResolvedExtensions<
+    UserPreferencesItem
+  >(isUserPreferencesItem);
+  const sortedUserPreferencesItems = orderExtensionBasedOnInsertBeforeAndAfter<
+    ResolvedUserPreferencesItem
+  >(userPreferencesItemExtensions.map(({ properties }) => properties));
+
+  // fetch the default category from the url params if available
+  const {
+    params: { category: groupIdFromUrl },
+  } = match;
+  const { id: initialTabId } =
+    sortedUserPreferencesGroups.length > 0
+      ? sortedUserPreferencesGroups.find((extension) => extension.id === groupIdFromUrl) ||
+        sortedUserPreferencesGroups[0]
+      : { id: '' };
+  const [activeTabId, setActiveTabId] = React.useState<string>(initialTabId);
+
+  // utils and callbacks
+  const getTabs = (): [React.ReactElement<TabProps>[], React.ReactElement<TabContentProps>[]] => {
+    const populatedUserPreferencesGroups: UserPreferencesTabGroup[] = getUserPreferencesGroups(
+      sortedUserPreferencesGroups,
+      sortedUserPreferencesItems,
+    );
+    const [tabs, tabContents] = populatedUserPreferencesGroups.reduce(
+      (acc, currGroup) => {
+        const { id, label, items } = currGroup;
+        const ref = React.createRef<HTMLElement>();
+        acc[0].push(
+          <Tab
+            key={id}
+            eventKey={id}
+            title={<TabTitleText>{label}</TabTitleText>}
+            href={`${USER_PREFERENCES_BASE_URL}/${id}`}
+            tabContentId={id}
+            tabContentRef={ref}
+            data-test={`tab ${id}`}
+          />,
+        );
+        acc[1].push(
+          <TabContent
+            key={id}
+            eventKey={id}
+            id={id}
+            ref={ref}
+            hidden={id !== activeTabId}
+            data-test={`tab-content ${id}`}
+          >
+            <Form>{getTabContents(items)}</Form>
+          </TabContent>,
+        );
+        return acc;
+      },
+      [[], []],
+    );
+    return [tabs, tabContents];
+  };
+  const handleTabClick = (event: React.MouseEvent<HTMLElement>, eventKey: string) => {
+    if (isModifiedEvent(event)) {
+      return;
+    }
+    event.preventDefault();
+    setActiveTabId(eventKey);
+    history.replace(`${USER_PREFERENCES_BASE_URL}/${eventKey}`);
+  };
+
+  // render elements
+  const [userPreferencesTabs, userPreferencesTabContents] = getTabs();
+
+  return (
+    <>
+      <Helmet>
+        <title>{t('console-app~User Preferences')}</title>
+      </Helmet>
+      <PageLayout
+        title={t('console-app~User Preferences')}
+        hint={t(
+          'Set your individual preferences for the console experience. Any changes will be autosaved.',
+        )}
+        isDark
+      >
+        {userPreferencesItemResolved ? (
+          <div className="ocs-user-preferences-content">
+            <div id="tabs-section" className="ocs-user-preferences-content__tabs">
```

Comment:
Can we remove this `id` and the one below?

Reply:
removed